default.nix

```diff
@@ -10,12 +10,14 @@
 
 , rev ? "29d57de30101b51b016310ee51c2c4ec762f88db" #  2020-05-23: NOTE: UTC 17:00
 
+, sha256 ? "1wjljkffb3gzdvpfc4v98mrhzack6k9i7860n8cf5nipyab6jbq9"
```

Comment:
Noob question: Where do I find the `sha256` for a particular nixpkgs revision?

Reply:
There are two easy ways: Using `nix-prefetch-url - -unpack`, or trying to build with an incorrect one.

Reply:
Thanks! :)

Reply:
Yep, but there is still that inhumane `sha256:` next to it.
...

The important thing to know that Nix ("somehow") prefers `sha` search over `rev`/version search. Version information concidered irrelevant meta information. If you update a version, and not updated a sha - Nix would successfully be happy to take that sha from CDN and nicely compile/package it, and wrap in beautiful inconsistent version information around it.

That is why we type a new version and then, invalidate the `sha` to get the `sha` to put the `sha`.

Thats is such a beautiful design right there.

Reply:
The thing is, git hashes are opaque to Nix and you would not be able to design a system of similar complexity for which they wouldn't be. You may be interested in https://discourse.nixos.org/t/obsidian-systems-is-excited-to-bring-ipfs-support-to-nix/7375, as that should enable Nix to verify build outputs using IPLD.

Reply:
Yep, Nix craves for IPFS integration.

I do not see why to not just get the secure cache from the central store, fetch the package from wherever it would come, and check it.